src/articles/index.tsx

```diff
@@ -15,25 +16,94 @@ import { Header } from '../misc/header';
 
 export function Articles(_: unknown, renderer: RendererLike<Node>) {
   return <>
-    <Route path='**/unapproved' comp={() =>
-      <ArticleList title='Unapproved Articles'
-        articles={fromPromise(getUnapprovedArticles(authToken()!))}
+    <Route path='**/unapproved' comp={() => {
+      const articles = state<Article[]>([]);
+      const isLoading = state(true);
+
+      const getArticles = (lastId?: string) => {
```

Comment:
since `<ArticleList/>` initiates loading more on its own, I suspect it is cleaner if it is passed a function instead of a `isLoading`, so that it can handle lazy loading completely on its own. this change would mean `lastId` also moves to `<ArticleList/>` component itself.

generally, _awareness_ of pagination mechanism and lazy loading is now split between two components. the functionality needs to be encapsulated in one area, so the `<ArticleList/>` component for example.

Reply:
Check the way I did it and tell me if it responds to this specific piece of feedback.